service/matching/matching_engine_test.go

```diff
@@ -2761,6 +2761,464 @@ func (s *matchingEngineSuite) TestUpdatePhysicalTaskQueueGauge_BuildID() {
 
 }
 
+func (s *matchingEngineSuite) TestAddConsumeWorkflowTasksValidateBacklogCount() {
+	s.matchingEngine.config.LongPollExpirationInterval = dynamicconfig.GetDurationPropertyFnFilteredByTaskQueueInfo(10 * time.Millisecond)
+
+	runID := uuid.NewRandom().String()
+	workflowID := "workflow1"
+	workflowType := &commonpb.WorkflowType{
+		Name: "workflow",
+	}
+	workflowExecution := &commonpb.WorkflowExecution{RunId: runID, WorkflowId: workflowID}
+
+	const taskCount = 100
+	// TODO: Understand why publish is low when rangeSize is 3
```

Comment:
The test passes even if we do have a small RangeSize and a large number of tasks. The check for "how many times Update Task Queue" was called had to change since it is called: `taskCount/configSize` number of times.